crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -485,14 +545,25 @@ fn write_pointer_lockfile(
     floxmeta: &FloxmetaV2,
     lock_path: PathBuf,
 ) -> Result<GenerationLock, EnvironmentError2> {
+    let local_ref = branch_name(system, pointer, lock_path.parent().unwrap()).unwrap();
     let remote_branch = remote_branch_name(system, pointer);
+
     let rev = floxmeta
         .git
         .branch_hash(&remote_branch)
         .map_err(ManagedEnvironmentError::GitBranchHash)?;
+
+    let local_rev = match floxmeta.git.branch_hash(&local_ref) {
+        Ok(local_rev) if local_rev == rev => None,
+        Ok(local_ref) => Some(local_ref),
+        // todo: think
```

Comment:
nonblocking: fill out this todo so someone else can understand it

Reply:
🙃 sure

Question to you would be whether you think "always" checking and updating this local_ref here makes sense and is logically correct or whether we need a "lock_local" equivalent?

Reply:
I'm basically trying to understand the locking workflow.
Lets sync on that tomorrow, the fact that there are ~10 tests for this tells me there is more to think about reg. this.